src/core/messaging/handlers/notification.ts

```diff
@@ -45,6 +46,22 @@ const handleNotification = (
         case NotificationType.TRANSACTION:
             const data = JSON.parse(notification.data);
 
+            let chainId = data.chainId;
+
+            // TODO: remove this on chainId refactor
+            switch (data.blockchain) {
```

Comment:
we have to avoid this kind of `switch`. I found the real problem :), we had it on Zilliqa too. The fix has to be done in `src/core/blockchain/ethereum/index.ts` in `getClient` function (check the solution on Zilliqa).

Reply:
Yes, nice, I had remember and updated :)